libs/langchain-community/src/vectorstores/pgvector.ts

```diff
@@ -677,4 +677,55 @@ export class PGVectorStore extends VectorStore {
     this.client?.release();
     return this.pool.end();
   }
+
+  /**
+   * Method to create the HNSW index on the vector column.
+   *
+   * @param dims - Defines the number of dimensions in your vector data, max: 2000. For example, use 1536 for OpenAI's text-embedding-ada-002 model and 1024 for amazon.titan-embed-text-v2:0
+   * @param m - The max number of connections per layer (16 by default)
+   * @param efConstruction -  The size of the dynamic candidate list for constructing the graph (64 by default)
+   * @param distanceFunction -  The distance function name you want to use, is automatically selected based on the distanceStrategy.
+   * @returns Promise that resolves with the query response of creating the index.
+   */
+  async createHnswIndex(config?: {
+    dims?: number;
+    m?: number;
+    efConstruction?: number;
+    distanceFunction?: string;
+  }): Promise<void> {
+    let idxDistanceFunction = config?.distanceFunction || "vector_cosine_ops";
+
+    switch (this.distanceStrategy) {
+      case "cosine":
+        idxDistanceFunction = "vector_cosine_ops";
+        break;
+      case "innerProduct":
+        idxDistanceFunction = "vector_ip_ops";
+        break;
+      case "euclidean":
+        idxDistanceFunction = "vector_l2_ops";
+        break;
+      default:
+        throw new Error(`Unknown distance strategy: ${this.distanceStrategy}`);
+    }
+
+    const createIndexQuery = `CREATE INDEX IF NOT EXISTS ${
+      this.vectorColumnName
+    }_embedding_idx
+        ON ${this.computedTableName} USING hnsw ((${
+      this.vectorColumnName
+    }::vector(${config?.dims || 1536})) ${idxDistanceFunction})
```

Comment:
Do we have any utilities to sanitize interpolated values? This code should be only exposed to developers, but if we have existing sanitization code won't hurt to add some more defensive code

Reply:
I've had the `pg-format` lib https://www.npmjs.com/package/pg-format to format the sql literals.
It should provide some guards for this.

Reply:
Can we actually remove this (for now)? It'll make existing users install a new package.

In the near-ish term, we should factor out into a `@langchain/pgvector` package that can have this as a hard dep.